apps/desktop/src/components/v3/NewCommit.svelte

```diff
@@ -39,7 +41,30 @@
 
 	const baseSha = $derived($base?.baseSha);
 	const defaultParentId = $derived(commit ? commit.id : baseSha);
-	const parentId = $derived(commitId ? commitId : defaultParentId);
+
+	const parentId = $derived.by(() => {
+		// If commitId is explicitly provided, use it
+		if (commitId) return commitId;
+
+		// Try to find parent based on branch position in stack
+		if (branches?.length) {
+			const currentBranchIndex = branches.findIndex((b) => b.name === branchName);
+
+			// If this branch has a "parent" branch in the stack
+			if (currentBranchIndex >= 0 && currentBranchIndex < branches.length - 1) {
+				const parentBranch = branches[currentBranchIndex + 1];
+				if (parentBranch?.name) {
+					const parentCommit = stackService.commitAt(projectId, stackId, parentBranch.name, 0)
+						.current.data;
+
+					if (parentCommit?.id) return parentCommit.id;
+				}
+			}
+		}
```

Comment:
Not sure if I'm reading correctly, but would this work if there are two empty branches, and you want to commit to the top one?

Reply:
In theory we could have any number of empty branches in between the one you're committing to and a previous commit (or base), so I'd prefer to avoid the looping altogether by making the API capable of committing directly to a branch without specifying the parent commit. Let's chat in person during once Kiril is in the office.

Reply:
I think the API already takes a branch name that is currently used to insert into the correct 'bucket'. From there the parent commit could trivially be extracted.

A branch name will always be known, but it won't have to necessarily be pointing to a commit within the stack that the user sees. The V3 workspace information can represent such cases, but for now it should be sufficient if the UI passes the name it knows, or the parent commit, or both, the backend should be able to deal with it.